### PR TITLE
fix(mqtt): Publish mosquitto bridge status to c8y when enabled

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

If the mosquitto bridge is used for C8y connectivity, then the bridge service status must also be published to C8y.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2898

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by rina23q — APPROVED** [review]
This PR fixes the reported bug

File: crates/extensions/c8y_mapper_ext/src/converter.rs [hunk 1]
```diff
@@ -359,7 +359,7 @@ impl CumulocityConverter {
                 let ancestors_external_ids =
                     self.entity_store.ancestors_external_ids(entity_topic_id)?;
 
-                if entity_topic_id.is_bridge_health_topic() {
+                if self.config.bridge_in_mapper && entity_topic_id.is_bridge_health_topic() {
                     return Ok(vec![]);
                 }
```
Comment (didier-wenzek) [inline crates/extensions/c8y_mapper_ext/src/converter.rs:364]: Adding a comment would to understand what it means to return an empty vector of messages.

```suggestion
                if self.config.bridge_in_mapper && entity_topic_id.is_bridge_health_topic() {
                    // Ignore bridge health topics when using the mapper-provided bridge
                    return Ok(vec![]);
                }
```

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by gligorisaev — COMMENTED** [review]
The test script is well-structured to verify the existence of an external identity and the status of the bridge service.

**Pull request merged**